src/sentry/static/sentry/app/views/sharedGroupDetails/index.tsx

```diff
@@ -1,130 +1,136 @@
+import {RouteComponentProps} from 'react-router/lib/Router';
 
 import {t} from 'app/locale';
+import {Client} from 'app/api';
 import withApi from 'app/utils/withApi';
 import {BorderlessEventEntries} from 'app/components/events/eventEntries';
 import Footer from 'app/components/footer';
 import LoadingError from 'app/components/loadingError';
 import LoadingIndicator from 'app/components/loadingIndicator';
 import NotFound from 'app/components/errors/notFound';
-import SentryTypes from 'app/sentryTypes';
-import SharedGroupHeader from 'app/views/sharedGroupDetails/sharedGroupHeader';
 import space from 'app/styles/space';
+import {Group} from 'app/types';
+import Link from 'app/components/links/link';
+import SentryTypes from 'app/sentryTypes';
+
+import SharedGroupHeader from './sharedGroupHeader';
 
-const SharedGroupDetails = createReactClass({
-  displayName: 'SharedGroupDetails',
+type Props = RouteComponentProps<{shareId: string}, {}> & {
+  api: Client;
+};
 
-  propTypes: {
-    api: PropTypes.object,
-  },
+type State = {
+  group: Group | null;
+  loading: boolean;
+  error: boolean;
+};
 
-  childContextTypes: {
+class SharedGroupDetails extends React.Component<Props, State> {
+  static childContextTypes = {
     group: SentryTypes.Group,
-  },
+  };
+
+  state: State = this.getInitialState();
 
   getInitialState() {
     return {
       group: null,
       loading: true,
       error: false,
     };
-  },
+  }
 
   getChildContext() {
     return {
       group: this.state.group,
     };
-  },
+  }
 
   componentWillMount() {
-    this.fetchData();
     document.body.classList.add('shared-group');
-  },
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
 
   componentWillUnmount() {
     document.body.classList.remove('shared-group');
-  },
+  }
+
+  async fetchData() {
+    const {params, api} = this.props;
+    const {shareId} = params;
+    try {
+      const group = await api.requestPromise(`/shared/issues/${shareId}/`);
+      this.setState({loading: false, group});
+    } catch {
+      this.setState({loading: false, error: true});
+    }
+  }
+
+  handleRetry = () => {
+    this.setState(this.getInitialState());
+    this.fetchData();
+  };
 
   getTitle() {
-    if (this.state.group) {
-      return this.state.group.title;
+    const {group} = this.state;
+
+    if (group) {
+      return group.title;
     }
-    return 'Sentry';
-  },
 
-  fetchData() {
-    this.setState({
-      loading: true,
-      error: false,
-    });
-
-    this.props.api.request(this.getGroupDetailsEndpoint(), {
-      success: data => {
-        this.setState({
-          loading: false,
-          group: data,
-        });
-      },
-      error: () => {
-        this.setState({
-          loading: false,
-          error: true,
-        });
-      },
-    });
-  },
-
-  getGroupDetailsEndpoint() {
-    const id = this.props.params.shareId;
-
-    return '/shared/issues/' + id + '/';
-  },
+    return 'Sentry';
```

Comment:
```suggestion
    return t('Sentry');
```

Reply:
I don't think we translate Sentry...do we?